### PR TITLE
B2 b 1533 accessand transfer destination tokens

### PR DESCRIPTION
**Pull request by afzalahmad-front**
## Issue ticket number and link from Jira

fixes: https://front-finance.atlassian.net/browse/B2B-1533

## Type of change

Please tick the relevant options (with an "x").

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] Refactor (none of the core functionality has changed)
- [ ] Chore (adding new tests, build processes, tooling, performance)
- [x] This change requires a documentation update

## Checklist:

Please tick the options (with an "x") of the actions which you have done. For example, it may be that an issue doesn't require new tests to be written, so you would leave option 5 unchecked:

- [x] I have performed a self-review of my code and it meets the criteria specified by the ticket/issue
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation if necessary
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have not changed existing tests (if so please add a description in the PR)
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: src/__tests__/__snapshots__/LinkConnect.test.tsx.snap [hunk 1]
```diff
@@ -12,11 +12,7 @@ exports[`LinkConnect Component renders correctly when linkToken is provided 1`]
   <View
     injectedJavaScript="
     window.meshSdkPlatform='reactNative';
-    window.meshSdkVersion='2.0.1';
-    if(window.parent){
-      window.parent.meshSdkPlatform='reactNative';
-      window.parent.meshSdkVersion='2.0.1';
-    }
+    window.meshSdkVersion='2.0.2';
```
Comment (vitaliimovchan-front) [inline src/__tests__/__snapshots__/LinkConnect.test.tsx.snap:15]: Are you going to change this snapshot each time the version is changing? Can we use some global variable here?

File: src/components/LinkConnect.tsx [hunk 2]
```diff
@@ -22,12 +22,29 @@ const LinkConnect = (props: LinkOptions) => {
...
     window.meshSdkPlatform='${sdkSpecs.platform}';
     window.meshSdkVersion='${sdkSpecs.version}';
-    if(window.parent){
-      window.parent.meshSdkPlatform='${sdkSpecs.platform}';
-      window.parent.meshSdkVersion='${sdkSpecs.version}';
-    }
   `;
 
+  const accessAndTransferTokensScript = useMemo(() => {
+    let script = '';
```
Comment (vitaliimovchan-front) [inline src/components/LinkConnect.tsx:28]: can we use single useMemo if we use the `sdkTypeScript` as initial here?

File: README.md [hunk 3]
```diff
@@ -89,12 +89,15 @@ export default App;
...
-| `onTransferFinished` | `(payload: TransferFinishedPayload) => void`    | optional                                  | Callback called when a crypto transfer is executed                            |
-| `onExit`             | `(err: string) => void)`                        | optional                                  | Called if connection not happened. Returns an error message                   |
+| key                         | type                                                                     | Required/Optional | description                                                                                    |
+|-----------------------------|--------------------------------------------------------------------------|-------------------|------------------------------------------------------------------------------------------------|
+| `linkToken`                 | `string`                                                                 | required          | link token                                                                                     |
```
Comment (vitaliimovchan-front) [inline README.md:94]:
```suggestion
| `linkToken`                 | `string`                                                                 | required          | Link token                                                                                     |
```

File: README.md [hunk 4]
```diff
@@ -89,12 +89,15 @@ export default App;
...
+| `accessTokens`              | `IntegrationAccessToken[]` (_check /lib/types/index.d.ts for reference_) | optional          | These access tokens are used to initialize crypto transfers flow at 'Select asset step'        |
+| `transferDestinationTokens` | `IntegrationAccessToken[]` (_check /lib/types/index.d.ts for reference_) | optional          | Can be used to initialize the crypto transfers flow as an alternative to the target addresses. |
```
[the remark this answers is not in the document]
Comment (vitaliimovchan-front) [reply README.md:96]: I'd suggest aligning the description with [iOS Readme](https://github.com/FrontFin/mesh-ios-sdk/pull/2/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5). Ask Jason to give the description and leave it on Notion.

File: src/types/index.ts [hunk 5]
```diff
@@ -55,8 +55,18 @@ export interface TransferFinishedErrorPayload {
...
 
+export interface IntegrationAccessToken {
+  accountId: string
+  accountName: string
+  accessToken: string
+  brokerType: string
+  brokerName: string
+}
+
 export interface LinkOptions {
```
Comment (AlexZeml-Front) [inline src/types/index.ts:66]: on iOS SDK we have the following

> public class LinkConfiguration {
>     var linkToken: String
>     var settings: LinkSettings? ...

> public struct LinkSettings {
>     public var useSecureOnDeviceStorage: Bool
>     public var accessTokens: [IntegrationAccessToken]?
>     public var transferDestinationTokens: [IntegrationAccessToken]?

can we make it consisten across all SDKs ?

File: src/__tests__/__snapshots__/LinkConnect.test.tsx.snap (hunk 1 above)
Comment (afzalahmad-front) [reply src/__tests__/__snapshots__/LinkConnect.test.tsx.snap:15]: actually this one is bind with the package.json file, so each time we update the version it should be changing the version in utility as well. 

Here' we always will get the version intended. (and snapshot is updated automatically)

File: src/types/index.ts (hunk 5 above)
Comment (afzalahmad-front) [reply src/types/index.ts]: I've harmonized the naming of settings and extra props.

File: README.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (vitaliimovchan-front) [reply README.md]: @afzalahmad-front Can you plz send me the link there this documentation takes place?

**Review by balint-front — APPROVED** [review]

**Pull request merged**